aten/src/ATen/core/dispatch/Dispatcher.cpp

```diff
@@ -102,6 +102,33 @@ RegistrationHandleRAII Dispatcher::registerDef(FunctionSchema schema) {
         op.operatorIterator_->op.registerSchema(std::move(schema));
       } else {
         TORCH_CHECK(op.schema() == schema, "Tried to register multiple operators with the same name and the same overload name but different schemas: ", schema, " vs ", op.schema());
+        // If the *new* schema is the default alias analysis kind, for BC, we
+        // will accept it.  If we don't accept it, most extensions that override
+        // existing operators will stop working (as they generally did not
+        // specify alias information).  Remove this BC smoothing ASAP, because
+        // if the two incompatible registrations live in the same compilation
+        // unit, the order their static initializers run is unspecified, which
+        // means that you may nondeterministically fail the subsequent test.
+        if (schema.isDefaultAliasAnalysisKind()) {
+          // OK
+        // If you POST-FACTO specify a non-default alias analysis kind after
```

Comment:
nit: indentation is weird. Maybe add the comments into the then/else block instead of before the block